### PR TITLE
Highlight search terms in results (closes #73, closes #115).

### LINKED ISSUE #73: Highlight search string matches in results list
screen shot 2015-11-19 at 7 31 23 am

Per mock. Note that it also matches in URLs.

### LINKED ISSUE #115: Recommended results are aren't treating <b> tags correctly
I guess they're being escaped somehow, because I'm seeing a lot of them:

### PR DESCRIPTION
**Pull request by chuckharmston**
screen shot 2015-12-06 at 3 50 37 pm

I tried to use [mustache lambdas](https://mustache.github.io/mustache.5.html#Sections) to achieve this, but I was never able to figure out how to get it to process a context variable; the lambdas were always passed the token, instead (e.g. `{{model.text}}`). This seems a fine substitute.

### CODE REVIEW

**Comment by jaredhirsch** [comment]
Actually, the mustache docs mention that the token is what's sent to the lambda function, so it needs to strip out the mustaches. If the lambda function is an arrow that executes in the view's context, then you should have a pointer to the view's model (if the problem is that model.foo is passed as a string, not as the value).

The current patch leaks rendering code into a model, where it should be confined to a view. You could set derived properties on the view itself, which could be calculated from the state of the model--however, in this case, I think a lambda is probably the way to go.

**Comment by chuckharmston** [comment]
Hmm...I inspected all the locals and wasn't able to find anything remotely resembling context. I must not have had the outer function as an arrow, nothing makes sense otherwise. I'll keep poking.

File: src/lib/template_helpers.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+export default {
+  stripHtml: () => {
+    // Strip all HTML from the passed string.
+    return (text, render) => {
+      let tmp = document.createElement('DIV');
+      tmp.innerHTML = text;
```
Comment (jaredhirsch) [inline src/lib/template_helpers.js:6]: @chuckharmston To address the specific bug here: `render(text)` is the value that you want.

Taking a step back, do we really need to do this at all? Hitting the DOM to filter text is a real performance hit, while the underlying issue is that we have some (accurately placed) bold tags coming back from the server. How about we just leave those `<b>` tags in there, for now, and plan in the future to have services return plain text, not html?

**Comment by jaredhirsch** [comment]
(Edited) eslint seems to be barfing on the unused helper import in the model.

Maybe we can debug together this afternoon :-)

**Comment by jaredhirsch** [comment]
It looks like current Nightly highlights search terms (see screenshot), so I'm going to close this in favor of figuring out what they're doing in https://github.com/mozilla/universal-search-addon/issues/191.

screenshot 2016-01-28 11 05 46

**Pull request closed without merging**